evalml/preprocessing/data_splitters/balanced_classification_splitter.py

```diff
@@ -89,4 +95,66 @@ def fit_resample(self, X, y):
+
+    def split(self, X, y):
+        """Splits and returns the labels of the training and testing data using the data sampler provided.
+        Arguments:
+                X (ww.DataTable): DataTable of points to split
+                y (ww.DataTable): DataColumn of points to split
+        Returns:
+            tuple(train, test): A tuple containing the resulting train and test indices, post sampling.
+        """
+        X_ww = infer_feature_types(X)
+        y_ww = infer_feature_types(y)
+        X = _convert_woodwork_types_wrapper(X_ww.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y_ww.to_series())
+        index_df = pd.Series(y.index)
+        for train, test in self.splitter.split(X, y):
+            X_train, y_train = X.iloc[train], y.iloc[train]
+            train_index_drop = self.sampler.fit_resample(X_train, y_train)
+            # convert the indices of the y column into index indices of the original pre-split y
+            train_indices = index_df[index_df.isin(train_index_drop)].dropna().index.values.tolist()
+            return iter([(train_indices, test)])
+
+
+class BalancedClassificationDataCVSplit(BaseUnderSamplingSplitter):
+    """Data splitter for generating k-fold cross-validation split using Balanced Classification Data Sampler."""
+
+    def __init__(self, balanced_ratio=4, min_samples=100, min_percentage=0.1, n_splits=3, shuffle=True, random_seed=0):
+        self.sampler = BalancedClassificationSampler(balanced_ratio=balanced_ratio, min_samples=min_samples, min_percentage=min_percentage, random_seed=random_seed)
+        super().__init__(sampler=self.sampler, n_splits=n_splits, random_seed=random_seed)
+        self.shuffle = shuffle
+        self.splitter = StratifiedKFold(n_splits=n_splits, shuffle=shuffle, random_state=random_seed)
+
+    def split(self, X, y):
+        """Splits and returns the sampled training data using the data sampler provided.
+        Arguments:
+                X (ww.DataTable): DataTable of points to split
+                y (ww.DataTable): DataColumn of points to split
+        Returns:
+            tuple(train, test): A tuple containing the resulting train and test indices, post sampling.
+        """
+        X_ww = infer_feature_types(X)
+        y_ww = infer_feature_types(y)
+        X = _convert_woodwork_types_wrapper(X_ww.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y_ww.to_series())
+        index_df = pd.Series(y.index)
+        for train, test in self.splitter.split(X, y):
+            X_train, y_train = X.iloc[train], y.iloc[train]
+            train_index_drop = self.sampler.fit_resample(X_train, y_train)
+            # convert the indices of the y column into index indices of the original pre-split y
+            train_indices = index_df[index_df.isin(train_index_drop)].dropna().index.values.tolist()
+            yield iter([train_indices, test])
```

Comment:
@chukarsten For CV cases, since split should be an iterable (over the number of splits), we want to return an iterator for the user so they can call `for train, test in s.split(X, y)`. If we `return` instead, we'll only get 1 split